zkevm-circuits/src/mpt_circuit/helpers.rs

```diff
@@ -1251,8 +1255,21 @@ impl<F: Field> MainRLPGadget<F> {
             // We enable dynamic lookups because otherwise these lookup would require a lot of extra
             // cells.
             cb.set_use_dynamic_lookup(true);
-            for (idx, byte) in config.bytes.iter().enumerate() {
-                require!((config.tag.expr(), byte.expr(), config.num_bytes.expr() - idx.expr()) => @FIXED);
+            if two_bytes_lookup {
+                for idx in 0..config.bytes.len() / 2 {
+                    let first = idx * 2;
+                    let second = idx * 2 + 1;
```

Comment:
I think you can just give in the index of the first byte as well. Because if you know the index of the 1st byte is `1`, then the index of the 2nd one is `0` which means the byte needs to be `0`.

Reply:
I was trying to group your original setup like you had `[tag, byte, len - idx]`, in which the last element could be for example 4, 3, 2, 1, 0, -1, -2, ..., -29 for len=4, or 5, 4, 3, ..., -28 when len=5
Then if i'm doing two bytes considering all possibilities it'll have 256*256*68*68 rows, so I group two indices by addition, namely 4, 3, 2, 1, 0, -1, -2, ..., -29 => 7, 3, 1, -3, ... and 5, 4 ,3, 2..., -28 => 9, 5, ...so that's why i need odd numbers in that range. But i guess we can do only the first byte because computing these indices itself doesn't constraint the actual byte location, it's just consistent with the original approach.

Reply:
Yeah it's just duplicated data, if we have idx for the first byte, we know the second byte will be `idx - 1`, we don't need the lookup to check that the index has been decreased by 1, that's just how it works.